src/autoscaler/eventgenerator/client/log_cache_client_test.go

```diff
@@ -53,28 +70,172 @@ var _ = Describe("LogCacheClient", func() {
 	})
 
 	JustBeforeEach(func() {
-		fakeLogCacheClient.ReadReturns(envelopes, logCacheClientReadError)
+		fakeGoLogCacheReader.ReadReturns(envelopes, logCacheClientReadError)
 		fakeEnvelopeProcessor.GetTimerMetricsReturns(metrics)
 		fakeEnvelopeProcessor.GetGaugeMetricsReturnsOnCall(0, metrics, nil)
 		fakeEnvelopeProcessor.GetGaugeMetricsReturnsOnCall(1, nil, errors.New("some error"))
 
-		logCacheClient = NewLogCacheClient(logger, func() time.Time { return collectedAt }, fakeLogCacheClient, fakeEnvelopeProcessor)
+		fakeGoLogCacheClient.WithViaGRPCReturns(expectedClientOption)
+
+		goLogCache := GoLogCache{
+			NewClient:   fakeGoLogCacheClient.NewClient,
+			WithViaGRPC: fakeGoLogCacheClient.WithViaGRPC,
+
+			WithHTTPClient:       fakeGoLogCacheClient.WithHTTPClient,
+			NewOauth2HTTPClient:  fakeGoLogCacheClient.NewOauth2HTTPClient,
+			WithOauth2HTTPClient: fakeGoLogCacheClient.WithOauth2HTTPClient,
+		}
+
+		logCacheClient.SetGoLogCache(goLogCache)
+		logCacheClient.Configure()
+	})
+
+	Context("NewLogCacheClient", func() {
+		BeforeEach(func() {
+			expectedAddrs = "logcache:8080"
+			logCacheClient = NewLogCacheClient(
+				logger, func() time.Time { return collectedAt },
+				fakeEnvelopeProcessor, expectedAddrs)
+
+		})
+
+		Context("when consuming log cache via grpc/mtls", func() {
+			var expectedTransportCredential credentials.TransportCredentials
+			var expectedDialOpt gogrpc.DialOption
+
+			BeforeEach(func() {
+				caCertFilePath = filepath.Join(testCertDir, "autoscaler-ca.crt")
+				certFilePath = filepath.Join(testCertDir, "eventgenerator.crt")
+				keyFilePath = filepath.Join(testCertDir, "eventgenerator.key")
+
+				expectedTlSCerts := &models.TLSCerts{KeyFile: keyFilePath, CertFile: certFilePath, CACertFile: caCertFilePath}
+				expectedTLSConfig, err := expectedTlSCerts.CreateClientConfig()
+				logCacheClient.SetTLSConfig(expectedTLSConfig)
+				expectedTransportCredential = credentials.NewTLS(expectedTLSConfig)
+				expectedDialOpt = gogrpc.WithTransportCredentials(expectedTransportCredential)
+				expectedClientOption = logcache.WithViaGRPC(expectedDialOpt)
+				fakeGRPC.WithTransportCredentialsReturns(expectedDialOpt)
+				Expect(err).NotTo(HaveOccurred())
+
+				grpc := GRPC{
+					WithTransportCredentials: fakeGRPC.WithTransportCredentials,
+				}
+				logCacheClient.SetGRPC(grpc)
+			})
+
+			JustBeforeEach(func() {
+				fakeGoLogCacheClient.WithViaGRPCReturns(expectedClientOption)
+			})
+
+			It("Should setup correct tls configurations", func() {
+				actualAddrs, actualClientOptions := fakeGoLogCacheClient.NewClientArgsForCall(0)
+
+				By("Creating the go log cache client with the correct params")
+				Expect(actualAddrs).To(Equal(expectedAddrs))
+				Expect(actualClientOptions).NotTo(BeEmpty())
+				Expect(reflect.ValueOf(actualClientOptions[0]).Pointer()).To(Equal(reflect.ValueOf(expectedClientOption).Pointer()))
```

Comment:
maybe have a look at how logcache client write their own tests around the options to see if there is a nice pattern we can re-use.